File: Striver/CPP/peak_element.cpp
#include <iostream>
#include <vector>

using namespace std;

int peak_element(vector<int> &nums) {
  int n = nums.size();
  if (n == 1) {
    return 0;
  }
  int low = 0;
  int high = n - 1;
  while (low < high) {
    int pivot = low + (high - low) / 2;
    if (nums[pivot] < nums[pivot + 1]) {
      low = pivot + 1;
    } else {
      high = pivot;
    }
  }
  return low;
}

int main() {
  vector<int> nums = {1, 2, 1, 3, 5, 6, 4};
  int peak = peak_element(nums);
  cout << "The peak element is : " << peak << endl;
}

File: Go/peak_element.go
package main
import "fmt";

func peakElement(nums []int) int{
	n := len(nums)
	if n==1{
		return 0
	}
	low := 0
	high := n-1
	for low<high {
		mid := low + (high - low)/2
		if nums[mid]<nums[mid+1] {
			low = mid+1
		} else{
			high = mid
		}
	}
	return low
}

func main(){
	nums := []int{1, 2, 1, 3, 5, 6, 4}
	peak := peakElement(nums)
	fmt.Println("The Peak Element Is : ", peak)
}